File: hw/basurto/u3/hw26Observer/hw26Observer.java/Observer/src/main/java/ec/edu/espe/Observer/model/Investor.java

package ec.edu.espe.Observer.model;
import ec.edu.espe.Observer.view.StockView;
/**
 *
 * @author Esteban Basurto, CodeBreakers, @ESPE
 */
public class Investor implements IInvestor {
    private String name;
    private String observerState;
    private Stock stock;

    public Investor(String name) {
        this.name = name;
    }

    @Override
    public void update(Stock stock, Object args) {
        this.stock = stock;
        StockView.printNotification(name, stock, args);
    }
}

File: hw/basurto/u3/hw26Observer/hw26Observer.py/model/investor.py
from model.i_investor import IInvestor
from view.stock_view import StockView

class Investor(IInvestor):
    def __init__(self, name: str):
        self.name = name
        self.observer_state = ""
        self.stock = None

    def update(self, stock, args):
        self.stock = stock
        StockView.print_notification(self.name, stock, args)